dev-client/src/persistence/kvStorage.ts

```diff
@@ -24,36 +30,61 @@ import {MMKVLoader} from 'react-native-mmkv-storage';
 // there's the underlying cross-platform native library:
 //   https://github.com/Tencent/MMKV
 // and then there's the react native wrapper library:
-//   https://www.npmjs.com/package/react-native-mmkv-storage
-// there is also a competing react native wrapper library:
 //   https://www.npmjs.com/package/react-native-mmkv
-//
-// the library we use simplifies encryption by automatically
-// generating an encryption key and storing it on the keychain,
-// which we would have to do manually if we were using the other
-// library. at time of writing there is not a well-documented
-// migration story from one to the other.
+// there is also a competing react native wrapper library, which
+// we used originally but abandoned due to slow maintenance:
+//   https://www.npmjs.com/package/react-native-mmkv-storage
 //
 // MMKV does not necessarily write to disk after every operation,
 // it will keep data in memory and only flush it when necessary
 // for performance reasons. I'm not sure if the RN API exposes a
 // way to manually flush to disk, there's a `clearMemoryCache`
 // method which presumably does this but haven't confirmed.
 //
-// `withEncryption` only has an effect when the device has never
-// initialized the MMKV store before. after first usage, the store
-// gets initialized with whatever settings it previously used.
-// there are additional methods available for changing the encryption
-// settings later: https://rnmmkv.vercel.app/#/workingwithencryption
-//
 // There is discussion on all of the libraries of a potential memory
 // use issue:
 //   https://github.com/ammarahm-ed/react-native-mmkv-storage/issues/286
 //   https://github.com/mrousavy/react-native-mmkv/issues/440
 //   https://github.com/Tencent/MMKV/issues/610
 // I ran a test in our environment and was unable to reproduce (persisted
 // entire redux state to storage in a loop for several hours), but leaving
-// here for context. There is also a `clearMemoryCache` method on storage
+// here for context. There is also a `trim` method on storage
 // object which may help.
 
-export const kvStorage = new MMKVLoader().withEncryption().initialize();
+const MMKV_STORAGE_ID = 'mmkv.storage';
+
+const mmkvStorage = new MMKV({id: MMKV_STORAGE_ID});
+
+export const kvStorage = {
+  setBool: (key: string, value: boolean) => mmkvStorage.set(key, value),
+  setString: (key: string, value: string) => mmkvStorage.set(key, value),
+  setNumber: (key: string, value: number) => mmkvStorage.set(key, value),
+  setObject: (key: string, value: object) =>
+    mmkvStorage.set(key, JSON.stringify(value)),
+  getBool: (key: string) => mmkvStorage.getBoolean(key),
+  getString: (key: string) => mmkvStorage.getString(key),
+  getNumber: (key: string) => mmkvStorage.getNumber(key),
+  getObject: <T extends object>(key: string) => {
+    const str = mmkvStorage.getString(key);
+    if (str === undefined) return str;
```

Comment:
question: why do we need setXXX/getXXX instead of the generic set/get the library provides?